src/geom/TileMatrix.js

```diff
@@ -0,0 +1,144 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * @exports TileMatrix
+ */
+define([
+        '../error/ArgumentError',
+        '../util/Logger',
+        '../geom/Sector'
+    ],
+    function (ArgumentError,
+              Logger,
+              Sector) {
+        "use strict";
+
+        /**
+         * TileMatrix is a collection of tiles for a fixed resolution (degrees per pixel). It contains the dimensions of
+         * the tile in pixels and the dimension of the matrix in tiles.
+         * @param sector the geographic coverage of this TileMatrix
+         * @param ordinal the index of this TileMatrix in the TileMatrixSet
+         * @param matrixWidth the number of tiles in the x direction
+         * @param matrixHeight the number of tiles in the y direction
+         * @param tileWidth the number of pixels or points in the x direction
+         * @param tileHeight the number of pixels or points in the y direction
+         * @constructor
+         */
+        var TileMatrix = function (sector, ordinal, matrixWidth, matrixHeight, tileWidth, tileHeight) {
+            if (!sector) {
+                throw new ArgumentError(
+                    Logger.logMessage(Logger.LEVEL_SEVERE, "TileMatrix", "constructor",
+                        "missingSector"));
+            }
+
+            if (ordinal < 0) {
+                throw new ArgumentError(
+                    Logger.logMessage(Logger.LEVEL_SEVERE, "TileMatrix", "constructor",
+                        "The specified oridinal is null or undefined."));
+            }
+
+            if (matrixWidth < 1) {
+                throw new ArgumentError(
+                    Logger.logMessage(Logger.LEVEL_SEVERE, "TileMatrix", "constructor",
+                        "invalidWidth"));
+            }
+
+            if (matrixHeight < 1) {
+                throw new ArgumentError(
+                    Logger.logMessage(Logger.LEVEL_SEVERE, "TileMatrix", "constructor",
+                        "invalidHeight"));
+            }
+
+            if (tileWidth < 1) {
+                throw new ArgumentError(
+                    Logger.logMessage(Logger.LEVEL_SEVERE, "TileMatrix", "constructor",
+                        "invalidWidth"));
+            }
+
+            if (tileHeight < 1) {
+                throw new ArgumentError(
+                    Logger.logMessage(Logger.LEVEL_SEVERE, "TileMatrix", "constructor",
+                        "invalidHeight"));
+            }
+
+            /**
+             * The geographic coverage area of this TileMatrix.
+             */
+            this.sector = sector;
+
+            /**
+             * The index position of this TileMatrix in the TileMatrixSet.
+             */
+            this.ordinal = ordinal;
```

Comment:
Could we use a more descriptive variable name here? LevelSet/Level uses levelNumber, but I'm not sure that's semantically correct here.

Reply:
This came from Android, but I don't recall whether or not it was necessary.

@zglueck Is ordinal used anywhere?

Reply:
@pdavidc it is used in the Tile key for TiledElevationCoverage

Reply:
Let's remove TileMatrix.ordinal. It was a shim that doesn't really belong. Additionally, I'll need to change how elevation tile keys are generated to accommodate the change in language.